### PR TITLE
729 stock line 2

### PR DESCRIPTION
**Pull request by mark-prins**
Another part of #729 

Updates the stock list f/e to use a specific endpoint and removes the client side data manipulation.
Wire up the item filter to run from the server instead of f/e - this was only a temporary implementation while the API didn't exist.

## Tests

- [x] There's now a `stockLines` query available in the graphQL schema
- [x] filtering by itemCodeOrName works, returns the correct result and count
- [ ] you can order stock lines by expiry date or soh
- [x] The stock list view is using the new query

### CODE REVIEW

File: server/repository/src/diesel_macros.rs [hunk 1]
```diff
@@ -95,6 +95,60 @@ macro_rules! apply_simple_string_filter {
...
+///     }
+///
+///     if let Some(value) = equal_filter.like {
+///         query = query.filter(item_dsl::code.like(format!("%{}%", value)));
+///         query = query.or_filter(item_dsl::name.like(format!("%{}%", value)));
+///     }
+/// }
+/// ```
+#[cfg(not(feature = "postgres"))]
+macro_rules! apply_simple_string_or_filter {
```
Comment (mark-prins) [inline server/repository/src/diesel_macros.rs:118]: bit of an offroad - adding in the ability to filter on two fields. have implemented for items (another offroad) and will wire that up in another PR for the item list

**Review by clemens-msupply — COMMENTED** [review]
When searching by a numeric code, 037 I get the error:

Error: Invalid value for argument "filter.itemCodeOrName.like", expected type "String"

**Comment by mark-prins** [comment]
> When searching by a numeric code, 037 I get the error:

As if anyone would search for a number in the code input 🙄 

Screenshot 2022-11-09 at 10 08 59 AM


..but thanks! this also fixes a problem in requisitions - the same thing happens. @bijaySussol spotted that one yesterday

f4aa66850

**Review by clemens-msupply — APPROVED** [review]
Looks great!

"you can order stock lines by expiry date or soh"

Isn't `stockOnHand = packSize * numberOfPacks`? Should there be a column for `packSize * numberOfPacks`?

**Comment by mark-prins** [comment]
🙏  for review @clemens-msupply! have added in the SOH column, and also enabled the column selection, now that there are a few to display!

**Comment by andreievg** [comment]
> @andreievg just seeing the conflicting file: server/schema.graphql do we need it in the repo or could we remove it and generate it locally when needed?

@clemens-msupply, I think it's similar to .generate.graphql files in front end (although might be more proned to conflict, not sure if it's always ordered correctly). It's used now for codegen of graphql `TS` types, it can be a hassle (conflicts can only be resolved by re-generating it after PR branch is updated from base branch), I think without it we have a risk of confusing with codegen. I personally think it's ok to keep, and if graphql schema is updated, we have to make sure to update from base branch and re-generate

**Comment by clemens-msupply** [comment]
Just checked again, think it is already autogenerated (from package.json): 

`"generate": "cd ../server && cargo run --bin remote_server_cli -- export-graphql-schema && cd ../client && graphql-codegen --config codegen.yml",`

**Comment by andreievg** [comment]
@clemens-msupply I thought that was a little bit slow and previously suggested just doing graphql-codegen for that script, but it would only be slow once, and @mark-prins pointed out the same as you, so it sounds like we should .gitignore it.

**Comment by mark-prins** [comment]
yeah - happy with either way. Probably least annoying to have it excluded, and generate each time it's required, it doesn't take very long.
I've had to regen every time I've used it, because there's always been a reason for generating, which is that there is new server code to generate against, and that's in an active branch. The alternative is to generate off the local server, which is what we did with programs feature, but I like the approach we have now, of building the server, exporting the schema and using that.

**Pull request merged**